submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestProcessor.kt

```diff
@@ -44,15 +49,25 @@ class SubmissionRequestProcessor(
         return processed
     }
 
-    private fun persistSubmissionFiles(sub: ExtSubmission) =
-        fileProcessingService.processFiles(sub) { file, idx ->
+    private fun persistSubmissionFiles(sub: ExtSubmission, startingAt: Int) {
+        fun persistSubmissionFile(file: ExtFile, idx: Int) {
             logger.info { "${sub.accNo} ${sub.owner} Started persisting file $idx, path='${file.filePath}'" }
 
             val persisted = storageService.persistSubmissionFile(sub, file)
+            val updatedFile = SubmissionRequestFile(sub.accNo, sub.version, idx, file.filePath, persisted)
+            filesRequestService.upsertSubmissionRequestFile(updatedFile)
             requestService.updateRequestIndex(sub.accNo, sub.version, idx)
 
             logger.info { "${sub.accNo} ${sub.owner} Finished persisting file $idx, path='${file.filePath}'" }
+        }
+
+        filesRequestService
+            .getSubmissionRequestFiles(sub.accNo, sub.version, startingAt)
+            .forEach { persistSubmissionFile(it.file, it.index) }
```

Comment:
I think this function need to be re structure to be more declarative. like it does the follow:

1. Load files
2. Save the file
3.  Update request File
4. Update index

we can express that as a single pipeline

```
        filesRequestService
            .getSubmissionRequestFiles(sub.accNo, sub.version, startingAt)
            .map { it.copy(file = storageService.persistSubmissionFile(sub, it.file)) }
            .onEach {
                filesRequestService.upsertSubmissionRequestFile(it)
                requestService.updateRequestIndex(sub.accNo, sub.version, it.index)
            }
```

Reply:
yes I agree it is more complex but also it is more self explanatory it depends to what do you want to give priority. Which of this solutions do you prefer? more details in https://medium.com/@jcamilorada/prefer-self-contained-code-the-lost-coding-principle-63020ee46b62 

![image](https://user-images.githubusercontent.com/28758892/199074564-9340b17d-6db4-4298-8152-891eefdd5cf6.png)